### PR TITLE
Refactor gceurlmap to a struct representation

### PR DESCRIPTION
**Pull request by rramkumar1**
Will fix tests once approach looks good.

/assign @nicksardo

### CODE REVIEW

**Comment by rramkumar1** [comment]
The tests in controller_test.go need a really big refactor so I am commenting out the portions which cause the unit tests to fail for now. I would like to really dig into fixing those tests in a separate PR.

**Comment by rramkumar1** [comment]
Decided to modify tests and fakes accordingly to make all tests work. Still want to dig deeper into controller_test.go later and fix it up.

File: pkg/controller/translator/translator.go [hunk 1]
```diff
@@ -71,15 +71,17 @@ type GCE struct {
...
+// ToURLMap converts an ingress to our internal UrlMap representation.
+func (t *GCE) ToURLMap(ing *extensions.Ingress) (*utils.GCEURLMap, error) {
+	urlMap := utils.NewGCEURLMap()
 	for _, rule := range ing.Spec.Rules {
 		if rule.HTTP == nil {
-			glog.Errorf("Ignoring non http Ingress rule")
 			continue
 		}
-		pathToBackend := map[string]string{}
+		// Using a mapping of regex -> backend here ensures that we don't
```
Comment (nicksardo) [inline pkg/controller/translator/translator.go]: Let's not use the word `regex` as this confuses users. Stick with path?

File: pkg/utils/gceurlmap.go [hunk 2]
```diff
@@ -18,45 +15,93 @@ package utils
...
+type GCEURLMap struct {
+	// DefaultBackendName is the k8s name given to the default backend.
+	DefaultBackendName string
+	// hostRules is a mapping from hostnames to path rules for that host.
+	hostRules map[string][]PathRule
+}
+
+// PathRule encapsulates the information for a single path -> backend mapping.
+type PathRule struct {
+	// Path is a regex.
```
Comment (nicksardo) [inline pkg/utils/gceurlmap.go]: sans regex

**Pull request merged**